backend/chromeApiServer/src/main/java/com/canvas/service/helperServices/HeaderFilter.java

```diff
@@ -0,0 +1,84 @@
+package com.canvas.service.helperServices;
+import java.io.IOException;
+import javax.servlet.Filter;
+import javax.servlet.FilterChain;
+import javax.servlet.ServletException;
+import javax.servlet.ServletRequest;
+import javax.servlet.ServletResponse;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletRequestWrapper;
+import javax.servlet.http.HttpServletResponse;
+import org.codehaus.plexus.util.StringUtils;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Component;
+
+/**
+ * Filter class for header processing.
+ */
+@Component
+public class HeaderFilter implements Filter {
+
+    /**
+     * Constructor for dependency injection of AESCryptoService.
+     * @param aesCryptoService instance of AESCryptoService
+     */
+    @Autowired
+    private final AESCryptoService aesCryptoService;
+
+    @Autowired
+    public HeaderFilter(AESCryptoService aesCryptoService) {
+        this.aesCryptoService = aesCryptoService;
+    }
+
+    /**
+     * Implements the doFilter method to process the header by implementing the filter class,
+     * all requests that come in go through this method.
+     *
+     * @param request servlet request
+     * @param response servlet response
+     * @param chain filter chain
+     * @throws IOException if an I/O error occurs
+     * @throws ServletException if a servlet error occurs
+     */
+    @Override
+    public void doFilter(ServletRequest request, ServletResponse response, FilterChain chain)
```

Comment:
update the name of the method to `executeFilter` or `runFilter`. Current name is too vague.

Reply:
References
https://www.baeldung.com/spring-boot-add-filter 
https://www.tutorialspoint.com/spring_boot/spring_boot_servlet_filter.htm
https://www.javadevjournal.com/spring-boot/spring-boot-add-filter/
doFilter is a common pattern in all springboot applications. let's stick to it.

Reply:
Ahh, this is overriding the existing function, got it